python/ray/tune/suggest/basic_variant.py

```diff
@@ -92,7 +102,7 @@ def _generate_trials(self, num_samples, unresolved_spec, output_path=""):
             raise TuneError("Must specify `run` in {}".format(unresolved_spec))
         for _ in range(num_samples):
             for resolved_vars, spec in generate_variants(unresolved_spec):
-                trial_id = "%05d" % self._counter
+                trial_id = self._uuid_prefix + ("%05d" % self._counter)
```

Comment:
I think this should be done in the `Trial.generate_id` codepath so that other trial-generating algorithms like Bayesian Optimization can also share this.

Reply:
I think the tradeoff is that then the counter wouldn't reset for each group of experiments.

Note that generate id already generates a UUID, this is just fixing the basic variant gen to have the hash in addition to the counter.

Reply:
ah got it